python/TestHarness/TestHarness.py

```diff
@@ -446,7 +446,7 @@ def printOutput(self, job, color):
         output = ''
         # Print what ever status the tester has at the time
         if self.options.verbose or (job.isFail() and not self.options.quiet):
-            output = 'Working Directory: ' + job.getTestDir() + '\nRunning command: ' + job.getCommand() + '\n'
+            output = 'Working Directory: ' + job.getTestDir() + '\nRunning command: ' + job.getCommand() + '\n' + ('Input File: ' + job.getTestDir() + job.getInputFile() if job.getInputFile() != None else 'No input file provided.')
```

Comment:
Is there a way we can use line continuation here, and/or use string replacement techniques?

eg:
```python
'Input File: %s' % (os.path.join(job.getTestDir(), job.getInputFile()))
or
'Input File: {}'.format(os.path.join(job.getTestDir(), job.getInputFile()))
```

Also, I think "Input File: None" would suffice here. As in, we might be able to find a clever way to just return None as a literal string and print that. Thoughts?

Reply:
I was probably the one that originally wrote that line of code :) before I knew about string formatting!

Reply:
I'm not sure I understand the point of the os.path calls, they just add runtime and the potential for unintended behavior. All this does is print a string of the input file path, where job.getTestDir() gets the path already through os.path, and job.getInputFile() just gets the name of the file. The rest of your changes are totally cosmetic, I don't particularly care if it says "No input file provided" or "Input file: None", it doesn't really matter, just let me know if you want it

Reply:
Without the os.path.join (using the system path separator) it constructs the path like this:

```
/Users/me/projects/moose/test/tests/test_harnessgood.i
```

when it should be:
```
/Users/me/projects/moose/test/tests/test_harness/good.i
```

We like to use Python's os.path.join, so we are not hard coding Unix-only path separators (for the eventual day we make this all work on Windows, we don't have to come back and switch a bunch of `/` for `\`).

As for 'Input File: None' vs 'No input file provided', your call. I am good either way. Upon first glance, I figured there was a way to do something like:

```python
output += 'Input File: %s' % (the path || str(job.getInputFile()))
```